tests/ffunctions/spin_fields/test_exchange_interaction_field.py

```diff
@@ -0,0 +1,71 @@
+        for j in range(num_neighbors[i]):
+            index = j + sum_nhbs
+            j_int = j_exchange[index]
+            nhb = neighbors[index]
+            total[i] -= j_int * state[nhb]
+    total /= spin_moments[:, numpy.newaxis]
+    return total
+
+
+@pytest.mark.repeat(100)
+def test_exchange_interaction_field_null_J_exchange(random_state_spins, build_sample):
+    num_sites, num_interactions, neighbors, num_neighbors = build_sample
+    spin_moments = numpy.ones(shape=num_sites)
+    j_exchange = numpy.zeros(shape=num_interactions)
+    total = numpy.zeros(shape=(num_sites, 3))
+    assert numpy.allclose(spin_fields.exchange_interaction_field(
+        random_state_spins, spin_moments, j_exchange, num_neighbors, neighbors), total)
+
+
+@pytest.mark.repeat(100)
+def test_exchange_interaction_field_constant_J_exchange(random_state_spins, build_sample):
+    num_sites, num_interactions, neighbors, num_neighbors = build_sample
+    spin_moments = numpy.ones(shape=num_sites)
+    j_exchange = numpy.full(num_interactions, numpy.random.uniform(-10, 10))
+    total = compute_exchange_field(
+        num_sites, random_state_spins, j_exchange, spin_moments, num_neighbors, neighbors)
+    assert numpy.allclose(spin_fields.exchange_interaction_field(
+        random_state_spins, spin_moments, j_exchange, num_neighbors, neighbors), total)
+
+
+@pytest.mark.repeat(100)
+def test_exchange_interaction_field_random_J_exchange(random_state_spins, build_sample, random_j_exchange):
+    num_sites, _, neighbors, num_neighbors = build_sample
+    spin_moments = numpy.ones(shape=num_sites)
+    total = compute_exchange_field(
+        num_sites, random_state_spins, random_j_exchange, spin_moments, num_neighbors, neighbors)
+    assert numpy.allclose(spin_fields.exchange_interaction_field(
+        random_state_spins, spin_moments, random_j_exchange, num_neighbors, neighbors), total)
+
+
+@pytest.mark.repeat(100)
+def test_exchange_interaction_field_random_spin_moments(random_state_spins, build_sample, random_spin_moments):
+    if numpy.any(random_spin_moments == 0.0):
+        raise ValueError("Magnetic moment must be non-zero")
+
+    num_sites, num_interactions, neighbors, num_neighbors = build_sample
+    j_exchange = numpy.full(num_interactions, numpy.random.uniform(-10, 10))
+    total = compute_exchange_field(
+        num_sites, random_state_spins, j_exchange, random_spin_moments, num_neighbors, neighbors)
+    assert numpy.allclose(spin_fields.exchange_interaction_field(
+        random_state_spins, random_spin_moments, j_exchange, num_neighbors, neighbors), total)
+
+
+@pytest.mark.repeat(100)
+def test_exchange_interaction_field_without_neighbors(random_state_spins, build_sample, random_j_exchange):
+    num_sites, num_interactions, _, _ = build_sample
+    spin_moments = numpy.ones(shape=num_sites)
+    num_neighbors = numpy.zeros(shape=num_sites, dtype=int)
+    neighbors = numpy.empty(shape=num_interactions, dtype=int)
```

Comment:
That is not what we want. We want something like:
```python
...
j_exchange = []
num_neighbors = [0] * num_sites
neighbors = []
...
```

Also, we don't need the `random_j_exchange` fixture here. You should remove it.